### PR TITLE
Build debs packaging

### PR DESCRIPTION
**Pull request by skarred14**
Enable caching of build debs.

Changes:
- New host directories for build-debs and toolchain-debs to store built deb packages, and corresponding changes to dockerfile, common variables and builder-utils.
- Update builder-api with additional helpers to invoke package builders: `package-invoke-builder` accepts arguments for SOURCE_PACKAGE_VERSION and package build/install controls.
- Build/install controls accept "bootstrap-only" (flag to store debs in toolchain-debs); "no-install" (flag to build deb but skip install)
- Introduction of new variable NON_PERSISTENT_ARCHIVE for bootstrap phases for gettext, util-linux and libxml2
- Introduction of a separate build stage to build binaries for bootstrapping packages: cuda, maven, go, and rust.
- Update package naming and build locations to use (BUILD)HOST_CPU instead of (BUILD)HOST_ARCH.
- Enable zstd compression.
- Enable stage root caching.

### CODE REVIEW

File: backends/bash/windows/val-verde-platform-sdk-swift-tools-windows [hunk 1]
```diff
@@ -6,16 +6,32 @@ set -e
...
+deb-package-mgmt ${PACKAGES_TO_MANAGE[@]}
+
+# install llvm for host
+PACKAGES_TO_MANAGE=( 'llvm-project-13' 'HOST')
+
+PACKAGE_ACTION=INSTALL \
+deb-package-mgmt ${PACKAGES_TO_MANAGE[@]}
 
 # souper build
 package-invoke-builder souper-cross
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-swift-tools-windows:24]: souper-cross missing version "1", i.e.
```
package-invoke-builder souper-cross 1
```

**Review by aaditya-chandrasekhar — COMMENTED** [review]
Souper builder fails on windows due to missing version.

File: backends/bash/windows/val-verde-platform-sdk-swift-corelibs-libdispatch-windows [hunk 2]
```diff
@@ -20,7 +16,10 @@ COMMON_CFLAGS="\
     -fms-extensions \
     ${COMMON_CFLAGS} \
 " \
-DISABLE_INSTALL_DEB=TRUE \
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-swift-corelibs-libdispatch-windows:23]: SWIFTCFLAGS line is reintroduced, this was removed.

File: backends/bash/windows/val-verde-platform-sdk-swift-corelibs-xctest-windows [hunk 3]
```diff
@@ -22,7 +18,11 @@ COMMON_CFLAGS="\
     -fms-extensions \
     ${COMMON_CFLAGS} \
 " \
-DISABLE_INSTALL_DEB=TRUE \
+PACKAGE_INSTALL_COMPLETION_CALLBACK=package-install-completion-callback \
+SWIFTCFLAGS="\
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-swift-corelibs-xctest-windows:22]: SWIFTCFLAGS line is reintroduced and was removed.

File: backends/bash/windows/val-verde-platform-sdk-yams-windows [hunk 4]
```diff
@@ -23,8 +19,11 @@ COMMON_CFLAGS="\
     -fms-extensions \
     ${COMMON_CFLAGS} \
 " \
-DISABLE_INSTALL_DEB=TRUE \
+SWIFTCFLAGS="\
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-yams-windows:22]: SWIFTCFLAGS is reintroduced here.

File: backends/bash/windows/val-verde-platform-sdk-swift-tools-support-core-windows [hunk 5]
```diff
@@ -23,9 +19,16 @@ COMMON_CFLAGS="\
     -fms-extensions \
     ${COMMON_CFLAGS} \
 " \
+<<<<<<< HEAD
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-swift-tools-support-core-windows:22]: Bad merge cruft.

File: backends/bash/windows/val-verde-platform-sdk-swift-tools-support-core-windows [hunk 6]
```diff
@@ -23,9 +19,16 @@ COMMON_CFLAGS="\
...
 DISABLE_INSTALL_DEB=TRUE \
+=======
+SWIFTCFLAGS="\
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-swift-tools-support-core-windows:25]: SWIFTCFLAGS reintroduced here.

**Review by aaditya-chandrasekhar — COMMENTED** [review]
Some merge problems in the following swift sdk components for windows:
1. libdispatch
2. foundation
3. tools-support-core
4. yams

File: backends/bash/windows/val-verde-platform-sdk-swift-corelibs-libdispatch-windows [hunk 7]
```diff
@@ -20,7 +16,10 @@ COMMON_CFLAGS="\
...
+SWIFTCFLAGS="\
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-swift-corelibs-libdispatch-windows:19]: SWIFTCFLAGS reintroduced, was removed.

File: backends/bash/windows/val-verde-platform-sdk-swift-corelibs-foundation-windows [hunk 8]
```diff
@@ -27,9 +23,13 @@ COMMON_CFLAGS="\
     -Wno-switch \
     ${COMMON_CFLAGS} \
 " \
-DISABLE_INSTALL_DEB=TRUE \
+PACKAGE_INSTALL_COMPLETION_CALLBACK=package-install-completion-callback \
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-swift-corelibs-foundation-windows:26]: PACKAGE_INSTALL_COMPLETION_CALLBACK was also removed.

File: backends/bash/windows/val-verde-platform-sdk-swift-corelibs-xctest-windows [hunk 9]
```diff
@@ -22,7 +18,11 @@ COMMON_CFLAGS="\
     -fms-extensions \
     ${COMMON_CFLAGS} \
 " \
-DISABLE_INSTALL_DEB=TRUE \
+PACKAGE_INSTALL_COMPLETION_CALLBACK=package-install-completion-callback \
```
Comment (aaditya-chandrasekhar) [inline backends/bash/windows/val-verde-platform-sdk-swift-corelibs-xctest-windows:21]: PACKAGE_INSTALL_COMPLETION_CALLBACK was removed.

Comment (skarred14) [reply backends/bash/windows/val-verde-platform-sdk-swift-corelibs-xctest-windows]: removed

File: backends/bash/windows/val-verde-platform-sdk-swift-corelibs-foundation-windows (hunk 8 above)
Comment (skarred14) [reply backends/bash/windows/val-verde-platform-sdk-swift-corelibs-foundation-windows]: removed

File: backends/bash/windows/val-verde-platform-sdk-swift-corelibs-libdispatch-windows (hunk 7 above)
Comment (skarred14) [reply backends/bash/windows/val-verde-platform-sdk-swift-corelibs-libdispatch-windows]: removed

File: backends/bash/windows/val-verde-platform-sdk-swift-tools-support-core-windows (hunk 6 above)
Comment (skarred14) [reply backends/bash/windows/val-verde-platform-sdk-swift-tools-support-core-windows]: fixed

File: backends/bash/windows/val-verde-platform-sdk-swift-tools-support-core-windows (hunk 5 above)
Comment (skarred14) [reply backends/bash/windows/val-verde-platform-sdk-swift-tools-support-core-windows]: fixed

File: backends/bash/windows/val-verde-platform-sdk-yams-windows (hunk 4 above)
Comment (skarred14) [reply backends/bash/windows/val-verde-platform-sdk-yams-windows]: removed

File: backends/bash/windows/val-verde-platform-sdk-swift-corelibs-xctest-windows (hunk 3 above)
Comment (skarred14) [reply backends/bash/windows/val-verde-platform-sdk-swift-corelibs-xctest-windows]: removed

**Pull request merged**